packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -261,8 +261,10 @@ library LibProposing {
             }
 
             // Check that after hooks, the Taiko Token balance of this contract
-            // have increased by at least config.livenessBond
-            if (tko.balanceOf(address(this)) < tkoBalance + config.livenessBond) {
+            // have increased by the same amount as config.livenessBond (to prevent)
+            // multiple draining payments by a malicious proposer nesting the same
+            // hook.
+            if (tko.balanceOf(address(this)) != tkoBalance + config.livenessBond) {
```

Comment:
check TKO balance change strict using == rather than >=.

Reply:
@adaki2004  with [this change](https://github.com/taikoxyz/taiko-mono/pull/15492/files), maybe we don't have to check "==" here?

Reply:
I still recommend having a strict equality, for two reasons
a) I don't see a benefit for transferring more than the liveness bond, it would increase the cost to the prover and the extra funds would be stuck in the TaikoL1 contract.
b) If multiple hooks are available which can transfer TKO to the TaikoL1 contract the proposer could attempt to use both e.g. If there is a third party hook, a proposer could attempt to use both hooks.

Reply:
> I still recommend having a strict equality, for two reasons

I think this is a strict equalitsy, no ? (if NOT -> revert)
`if (tko.balanceOf(address(this)) != tkoBalance + config.livenessBond) revert L1_LIVENESS_BOND_NOT_RECEIVED`

Reply:
Yep this PR currently has a strict equality, which I currently see as a nicer solution. I was responding to @dantaik comment about potentially moving back to inequality.
 
 > @adaki2004 with [this change](https://github.com/taikoxyz/taiko-mono/pull/15492/files), maybe we don't have to check "==" here?